generator/testdata/google/example/library/v1/library_service.gapic.cc.baseline

```diff
@@ -9,84 +11,192 @@
 gax::StatusOr<::google::example::library::v1::Book>
 LibraryService::CreateBook(
 ::google::example::library::v1::CreateBookRequest const& request) {
-  // TODO: actual useful work, e.g. retry, backoff, metadata, pagination, etc.
+  // TODO: update metadata and idempotency, etc.
   grpc::ClientContext context;
   ::google::example::library::v1::Book response;
   grpc::Status status = stub_->CreateBook(&context, request, &response);
+
   if (status.ok()) {
     return response;
-  } else {
-    return gax::GrpcStatusToGaxStatus(status);
   }
+
+  std::string error_message;
+  // TODO: use idempotency policy to determine whether a failure is permanent
+  if (!isPermanentFailure(status)) {
+    auto retry_policy = retry_policy_->clone();
+    auto backoff_policy = backoff_policy_->clone();
+    while (retry_policy->OnFailure(status)) {
+      auto backoff_duration = backoff_policy->OnCompletion();
+      std::this_thread::sleep_for(backoff_duration);
+      status = stub_->CreateBook(&context, request, &response);
+      if (status.ok()) {
+        return response;
+      }
+    // TODO: augment error_message with result from status
```

Comment:
I don't think we need to augment the error message here, just return a direct conversion from the grpc::Status to gax::Status. Or, perhaps I should say - what additional info from each status would we add to the message? I'm not sure at this point.

Reply:
cloud-cpp augments the error message based on whether call is idempotent or if the error is permanent. It complicates the logic of the retry loop, and I don't know how useful that would actually be to someone looking at a log message. I'm personally against it but left the option open.